### PR TITLE
test: Add GraphQL field resolution & validation tests

### PR DESCRIPTION
**Pull request by pekkah**
## Summary
- Comprehensive field resolution and validation tests covering edge cases and scenarios
- Tests field resolution with various argument types: strings, numbers, lists, enums, input objects
- Tests nested field resolution, interface/union field handling, and custom scalar arguments
- Tests resolver error scenarios, null handling, and non-null field validation
- Tests variable argument handling, validation edge cases, and deprecated field access
- Covers comprehensive field resolution patterns and validation error conditions

## Test plan
- [x] Builds successfully with proper ResolversMap syntax and API usage
- [x] Comprehensive test coverage of field resolution scenarios
- [x] Uses correct Executor.Execute API patterns and resolver implementations
- [ ] All test cases pass (some test expectations may need adjustment based on ValidationException behavior)
- [ ] Tests provide comprehensive coverage of field resolution and validation edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/GraphQL.Tests/FieldResolutionValidationFacts.cs [hunk 1]
```diff
@@ -0,0 +1,902 @@
...
+
+        var query = @"{ 
+            processUser(input: { name: ""John"", age: 30 }) 
+        }";
+
+        var result = await Executor.Execute(schema, query);
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Data);
+        Assert.Equal("Processed user: John (30) - noemail@example.com", result.Data["processUser"]);
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/FieldResolutionValidationFacts.cs:162]: ✅ **Fixed**: Added `Assert.Null(result.Errors);` assertion to verify no errors were returned, making the test more robust.

File: tests/GraphQL.Tests/FieldResolutionValidationFacts.cs [hunk 2]
```diff
@@ -0,0 +1,902 @@
...
+        }";
+
+        var result = await Executor.Execute(schema, query);
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Data);
+        Assert.Null(result.Errors);
+
+        var animals = (List<object>)result.Data["animals"];
+        Assert.Equal(2, animals.Count);
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/FieldResolutionValidationFacts.cs:323]: ✅ **Fixed**: Added comprehensive assertions to verify all resolved object properties:
- Verified animals list has 2 items
- Verified Dog object: name="Buddy", sound="Woof", breed="Golden Retriever"
- Verified Cat object: name="Whiskers", sound="Meow", color="Orange"

This makes the interface test much more valuable by ensuring proper polymorphic resolution.

File: tests/GraphQL.Tests/FieldResolutionValidationFacts.cs [hunk 3]
```diff
@@ -0,0 +1,902 @@
...
+                    context.ResolvedValue = $"Hello, {name}!";
+                    return ValueTask.CompletedTask;
+                }},
+                { "greetingWithDefault", context =>
+                {
+                    var name = context.ArgumentValues.TryGetValue("name", out var nameValue)
+                        ? (string)nameValue
+                        : "Anonymous";
+                    context.ResolvedValue = $"Hello, {name}!";
+                    return ValueTask.CompletedTask;
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/FieldResolutionValidationFacts.cs]: ✅ **Fixed**: Simplified the `greetingWithDefault` resolver by removing duplicated default value logic. The schema already defines `name: String = "Anonymous"` so the resolver now simply uses the provided value directly, eliminating redundancy.

File: tests/GraphQL.Tests/FieldResolutionValidationFacts.cs [hunk 4]
```diff
@@ -0,0 +1,902 @@
...
+        var resolvers = new ResolversMap
+        {
+            ["Query"] = new()
+            {
+                { "processUser", context =>
+                {
+                    var input = (Dictionary<string, object>)context.ArgumentValues["input"];
+                    var name = (string)input["name"];
+                    var age = (int)input["age"];
+                    var email = input.TryGetValue("email", out var emailValue)
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Tests/FieldResolutionValidationFacts.cs]: ✅ **Fixed**: Simplified the input object resolver by removing duplicated default value logic for the `email` field. The schema already defines `email: String = "noemail@example.com"` so the resolver now directly uses the provided value, eliminating redundancy.
